crates/node/events/src/node.rs

```diff
@@ -664,6 +671,21 @@ impl Display for Eta {
     }
 }
 
+/// Transforms a stream of `Result<T, BroadcastStreamRecvError>` into a stream of `NodeEvent`,
+/// applying a uniform error handling and conversion strategy.
+pub fn handle_broadcast_stream<T>(
+    stream: impl Stream<Item = Result<T, BroadcastStreamRecvError>> + Unpin,
+) -> impl Stream<Item = NodeEvent> + Unpin
+where
+    T: Into<NodeEvent>,
+{
+    stream.map(|result_event| {
+        result_event
+            .map(Into::into)
+            .unwrap_or_else(|err| NodeEvent::Other(format!("Stream error: {:?}", err)))
+    })
+}
+
```

Comment:
how about implementing `FromIterator` here, that will work I think

Reply:
> how about implementing `FromIterator` here, that will work I think

the map is provided by streamext, and streams are not iterators, so I'm not sure `Fromiterator` is the right fit here

Reply:
https://docs.rs/tokio-stream/latest/tokio_stream/trait.FromStream.html

Reply:
rip, would be nice but it's sealed, hopefully soon :tm: in stable